### PR TITLE
Implementing the `dot` BLAS and LAPACK Operation

### LINKED ISSUE #11917: dot

### PR DESCRIPTION
**Pull request by jatinS-dev**
close #11917

### CODE REVIEW

**Comment by jatinS-dev** [comment]
@Ishticode any feedback?

File: ivy/functional/frontends/torch/blas_and_lapack_ops.py [hunk 1]
```diff
@@ -193,3 +193,12 @@ def vdot(input, other, *, out=None):
         raise RuntimeError("input must be 1D vectors")
     input, other = torch_frontend.promote_types_of_torch_inputs(input, other)
     return ivy.vecdot(input, other, out=out)
+
+
+@to_ivy_arrays_and_back
+def dot(input, other, *, out=None):
+    if len(ivy.shape(input)) == 1 or len(ivy.shape(other)) == 1:
```
Comment (Ishticode) [inline ivy/functional/frontends/torch/blas_and_lapack_ops.py:200]: I think the torch dot function requires both input and other to be 1D not just one of them. So this should be `and` not `or`. Feel free to correct me.

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi @jatinS-dev 
Thank you for the PR.
First and foremost could you mention/link the issue that this PR will resolve/complete. Atm it seems `dot` is unassigned. Please check the instructions for assigning the task to yourself from [this section](https://lets-unify.ai/ivy/contributing/the_basics.html#todo-list-issues) of the docs and assign yourslef from the [BLAS and LAPACK ToDo list](https://github.com/unifyai/ivy/issues/17).

This methodology is necessary to keep track of all the functionalities being added.

I have left one comment under your implementation of the dot function.

You have also added `dot` in the `linalg` section with `dim` arg. I couldn't find this version in torch docs but even it exists, it would be a separate task as the additional param would mean a different operation with its own test. For now you can delete it from linalg.

Feel free to request review when done.
Thanks

**Comment by jatinS-dev** [comment]
PR for the issue #11917

File: ivy/functional/frontends/torch/blas_and_lapack_ops.py (hunk 1 above)
Comment (jatinS-dev) [reply ivy/functional/frontends/torch/blas_and_lapack_ops.py]: Yes, you are right. Both the input and the other should be 1D.

**Comment by jatinS-dev** [comment]
Thanks for the feedback! @Ishticode done the necessary changes.

**Review by Ishticode — APPROVED** [review]
Hi @jatinS-dev 
Looking great. I've resolved the conflicts and linked the issue. Will be getting this merged now.🙂
Thank you very much for the PR and you contribution. :grin:

**Pull request merged**